src/Microsoft.AspNetCore.Sockets.Abstractions/ConnectionBuilderExtensions.cs

```diff
@@ -1,16 +1,21 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using Microsoft.AspNetCore.Protocols;
-using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.AspNetCore.Sockets
 {
     public static class ConnectionBuilderExtensions
     {
         public static IConnectionBuilder UseEndPoint<TEndPoint>(this IConnectionBuilder connectionBuilder) where TEndPoint : EndPoint
         {
-            var endpoint = connectionBuilder.ApplicationServices.GetRequiredService<TEndPoint>();
+            var endpoint = (TEndPoint)connectionBuilder.ApplicationServices.GetService(typeof(TEndPoint));
```

Comment:
Where is no generic `GetService`?

Reply:
I didn't want to add a dependency on dependency injection 😄